### PR TITLE
STENCIL-3062 Upgrade stencil-styles

### PR DESCRIPTION
**Pull request by mcampa**
- Adding travis
- eslint
- cleaning code

@bigcommerce/stencil-team @lord2800

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+language: node_js
+sudo: false
+node_js:
+  - 4.4.7
```
Comment (lord2800) [inline .travis.yml]: Does it need to use this specific version of node? Would `4` be sufficient?

File: lib/styles.js [hunk 2]
```diff
@@ -85,13 +81,13 @@ module.exports = function StencilStyles() {
...
+        const sassFunctions = {};
 
-        sassFunctions['stencilNumber($name, $unit: px)'] = function(nameObj, unitObj) {
-            var name = nameObj.getValue(),
-                unit = unitObj.getValue(),
-                value = 0;
+        sassFunctions['stencilNumber($name, $unit: px)'] = (nameObj, unitObj) => {
+            const name = nameObj.getValue();
+            const unit = unitObj.getValue();
+            var value = 0;
```
Comment (lord2800) [inline lib/styles.js]: Can this be `let`, or are we dealing with too old of a version of node?

Comment (mcampa) [inline lib/styles.js]: old version of node :(

File: package.json [hunk 3]
```diff
@@ -19,13 +21,12 @@
   "homepage": "https://github.com/bigcommerce/stencil-styles#readme",
   "dependencies": {
     "autoprefixer-core": "^5.2.1",
-    "hoek": "^2.14.0",
-    "less": "^2.5.1",
```
Comment (lord2800) [inline package.json]: Hah!

File: lib/styles.js (hunk 2 above)
Comment (lord2800) [inline lib/styles.js]: Curses!

File: package.json (hunk 3 above)
Comment (mcampa) [inline package.json]: not used. we only compile sass ¯\\\_(ツ)\_/¯

Comment (lord2800) [inline package.json]: Which is why I was laughing. 😄

**Pull request merged**